### DISCUSSION TITLE
linux-service.json file changes are not reflected in UI

### DISCUSSION THREAD

**Discussion by muthumdu2k8** [apache/skywalking#10377] [Q&A]
I am working on Linux server monitoring. In dashboard i need to do some changes in UI (eliminate some of the metrics) and in values of the metrics from KB to MB and from MB to GB. 

For that i edited the /skywalking/config/ui-initialized-templates/os_linux/linux-service.json file. these changes are not reflected in UI.

i set the environment variable SW_ENABLE_UPDATE_UI_TEMPLATE: "true" and edited ui and saved. once i make the docker compose down and up the changes are not present.

please guide me what else i need to do for the above changes.

**Comment by wu-sheng** [comment]
This file is only being loaded once for the first time. You should either remove this from the database or use UI to setup and save the changes. Don't recommend to edit this file manually.

**Reply by muthumdu2k8** [reply]
thanks for your reply.

I agree your point, i kept the linux-service.json file as it is. i edited UI and saved the changes. for some reason i need to restart the oap server. After restarting oap server the changes i did in UI are lost. please help me on persisting the changes.

**Reply by wu-sheng** [reply]
The change would be saved if you choose. 

<img width="171" alt="image" src="https://user-images.githubusercontent.com/5441976/218093251-e7d9f8d6-6f20-4edb-ab11-f271e44f1d36.png">

Notice, UI should report by default, unless you opened this on OAP.
> The dashboard update has been disabled. Check SW_ENABLE_UPDATE_UI_TEMPLATE on configuration-vocabulary.md(https://skywalking.apache.org/docs/main/next/en/setup/backend/configuration-vocabulary/#configuration-vocabulary) to activate it

**Reply by muthumdu2k8** [reply]
I enabled this environment variable SW_ENABLE_UPDATE_UI_TEMPLATE: "true" and save is also done.
![image](https://user-images.githubusercontent.com/123181688/218373306-773704a4-d58e-45f3-a5b2-ac2c0b284c27.png)
after that when i restart the OAP server changes in UI are lost.

**Reply by wu-sheng** [reply]
Check the template after you saved. AFAIK, rebooting would not reset templates.

**Reply by wu-sheng** [reply]
Save should only from UI/GraphQL, rather than through JSON file.
Please make sure you are using latest and JSON in database(in memory H2 excluded) is changed.

**Reply by muthumdu2k8** [reply]
I am using PostgreSQL for storage. In ui-templates table, Row with id "Linux-service" the changes are saved in JSON format. when i am restarting both postgres and OAP that time UI changes are lost.

is there any way to persist the changes after restarting of both oap and postgres?

**Reply by wu-sheng** [reply]
@wankai123 Could you verify this?

**Reply by wankai123** [reply]
I have check with PostgreSQL storage: 
1. Edit some dashboards on UI and save them. -----> the JSON in PostgreSQL changed.
2. Restart OAP. ------> the JSON not override by initial config.
3. Refresh the UI with no-cache model, the changes still there.
So I think it works well.

**Reply by wu-sheng** [reply]
@muthumdu2k8 We have verified again, and it works for the latest version. Please check your local env, I think you either don't save the change in the storage, or there is a bug we can't reproduce.

**Reply by muthumdu2k8** [reply]
@wankai123 the steps you followed is correct that is working for me also. my case is once we restart both OAP and postgres the UI changes are lost.

**Reply by wankai123** [reply]
I think you should check your postgres whether lost data when restart.

**Reply by muthumdu2k8** [reply]
yes data is lost in postgres when i restart

**Reply by wu-sheng** [reply]
Then you should learn Postgresql deployment. We do touch database data.

**Comment by muthumdu2k8** [comment]
<img width="947" alt="image" src="https://user-images.githubusercontent.com/123181688/218383985-933b9ce6-bdcb-44f3-8373-62135d5ed8ab.png">
how to enable debug logs of OAP server?

**Reply by wu-sheng** [reply]
That is general log4j configuration. Nothing special.
